Lib/test/test_typing.py

```diff
@@ -3778,6 +3790,60 @@ def __init__(self):
 
         Foo()  # Previously triggered RecursionError
 
+    def test_get_protocol_members(self):
+        with self.assertRaises(TypeError):
+            get_protocol_members(object)
+        with self.assertRaises(TypeError):
+            get_protocol_members(object())
+        with self.assertRaises(TypeError):
+            get_protocol_members(Protocol)
```

Comment:
This test is failing because `TypeError` isn't being raised here at the moment, but I'm not sure it's essential that `TypeError` be raised here. Could just delete this assertion:

```suggestion
```

Reply:
I would like it to raise, let me see what's going on.

Reply:
It's because `Protocol` does actually have `__protocol_attrs__` defined, and I suggested you use a duck-type-y implementation where it just returns a copy of the `__protocol_attrs__` attribute if it exists

Reply:
I see, thanks. In that case I'd like to go back to checking `is_protocol()` explicitly in `get_protocol_members()`. I don't want to have to guarantee backward compatibility for how this function behaves on things that aren't Protocols.

Reply:
Sure 👍